node/src/block_producer/block_producer_effects.rs

```diff
@@ -151,6 +151,16 @@ pub fn block_producer_effects<S: crate::Service>(
                 .collect();
             // TODO(binier)
             let supercharge_coinbase = true;
+            // We want to know if this is a new epoch to decide which staking ledger to use
+            // (staking epoch ledger or next epoch ledger).
+            let is_new_epoch = won_slot.epoch()
```

Comment:
I am not entirely sure about these comparisons here, these slot/epoch values confuse me (they are are already confusing in the OCaml node, and in our case I think we have even more variants).

I am assuming the `WonSlot` values and the `epoch_count` in the consensus state is compatible.

@adonagy do you know if this is correct?

Reply:
ok just confirmed that both use the same point of reference (genesis vs hardfork) and it is ok to compare them.